### ISSUE TITLE
Recognize callback factory / currying

### ISSUE THREAD

**Issue by elsassph** [solidjs-community/eslint-plugin-solid#192]
**Describe the need**

It seems to be allowed to access `props` inside an `onSomething` function, e.g.
```typescript
const onSomething = (): void => {
  const duration = props.duration;
  console.log(duration);
}
```

However using a factory/currying pattern causes reactivity issues:
```typescript
// Warning: "This function should be passed to a tracked scope..."
const onSomething = (delay: number): void => () => {
  const duration = delay + props.duration;
  console.log(duration);
}
```

Even when attempting to name the inner function `onSomething` (not great anyway)
```typescript
// Warning: "This function should be passed to a tracked scope..."
const createSomethingHandler = (delay: number): void => 
  function onSomething(): void {
    const duration = delay + props.duration;
    console.log(duration);
  }
}
```

**Suggested Solution**

Recognize the pattern :)

**Possible Alternatives**

Sometimes it is possible to pass all the props at the creation point (which would happen safely in an effect/JSX), but it isn't always practical.

**Additional context**

- [x] I would be willing to contribute a PR to implement this feature

**Comment by ryansolid** [comment]
Verified fixed on current main: the callback-factory/currying pattern (`const onSomething = (delay) => () => { ...props... }`) no longer warns — returned functions are now recognized as the custom-primitive/handler-factory contract (#214). Ships in the next release.
